pkg/k8s/pod.go

```diff
@@ -414,25 +425,39 @@ func buildContainerVolumes(name string, volumes []*Volume, files []*File) []v1.V
 		)
 	}
 
-	var containerFiles []v1.VolumeMount
-
-	for n, file := range files {
-		shouldAddFile := true
-		for _, volume := range volumes {
-			if strings.HasPrefix(file.Dest, volume.Path) {
-				shouldAddFile = false
-				break
-			}
+	if len(volumes) == 0 && len(files) != 0 {
```

Comment:
what's the main difference between `buildContainerVolumes` and `buildInitContainerVolumes`? 

Feels like we should be able to refactor this to something like:

```go
func buildInitContainerVolumes() {
  // some code
 return buildContainerVolumes()
}
```

Reply:
maybe a good follow up. Looks like there are a bunch of functions in this file that could be structured in this way.

Reply:
Even if the functions look similar, they do not do the same. The init container is used to prepare files and volumes for the main container.

I agree that we need to improve file and volume handling.

Ref.:
- https://github.com/celestiaorg/knuu/issues/390
- https://github.com/celestiaorg/knuu/issues/529